types/oracle.go

```diff
@@ -0,0 +1,117 @@
+package types
+
+import "time"
+
+type AccountsOracle struct {
```

Comment:
Consider adding JSON struct tags to the fields to ensure compatibility and ease of use with JSON-based operations. [medium]